### ISSUE TITLE
Migrate to `styled-components`

### ISSUE THREAD

**Issue by Niyousha99** [techstartucalgary/tsu-website#287]
We are currenly following the old approach of using css stylesheets which is not ideal for global scope and dynamic styling. After some investigation, it would be more beneficial for our project to migrate to [styled-components](https://styled-components.com/), for meeting the previously mentioned requirements as well as creating a reusable design system with theming and using SASS syntax.

I propose we start with migrating our smallest components whose styling scope changes won't largely affect other areas in code.

**Comment by Niyousha99** [comment]
Remaining css modules to convert to styled components:

- [ ] App.css
- [ ] HomePage.css
- [ ] Team.css (TODO after #297 is done)
- [ ] DocsPage.css
- [ ] GuidesSection.css
- [ ] ApplyPage.css
- [ ] FeaturedProject.css
- [ ] Blobbie.css (low priority)
- [ ] Navbar.css
- [ ] NumberStat.css

**Comment by Niyousha99** [comment]
Rest of work will be covered in https://github.com/techstartucalgary/tsu-website/issues/494
